tests/examples.rs

```diff
@@ -209,3 +211,32 @@ fn double_choices() {
     let computed_solution = resolve(&dependency_provider, "a", 0).unwrap();
     assert_eq!(expected_solution, computed_solution);
 }
+
+#[test]
+fn confusing_with_lots_of_holes() {
+    let mut dependency_provider = OfflineDependencyProvider::<&str, NumVS>::new();
+
+    // root depends on foo...
+    dependency_provider.add_dependencies("root", 1, vec![("foo", Range::full())]);
+
+    for i in 1..6 {
+        // foo depends on bar...
+        dependency_provider.add_dependencies("foo", i as u32, vec![("bar", Range::full())]);
+    }
+
+    let Err(PubGrubError::NoSolution(mut derivation_tree)) =
+        resolve(&dependency_provider, "root", 1)
+    else {
+        unreachable!()
+    };
+    assert_eq!(
+        &DefaultStringReporter::report(&derivation_tree),
+        r#"Because there is no available version for bar and foo 1 | 2 | 3 | 4 | 5 depends on bar, foo 1 | 2 | 3 | 4 | 5 is forbidden.
+And because there is no version of foo in <1 | >1, <2 | >2, <3 | >3, <4 | >4, <5 | >5 and root 1 depends on foo, root 1 is forbidden."#
+    );
+    derivation_tree.collapse_no_versions();
```

Comment:
I'm not following why a test showcasing the effect of `derivation_tree.collapse_no_version()` is relevant in a PR where the behavior change happens while building the solution, and not while simplifying the derivation tree of the error. Do you mind reminding me why that is?

Reply:
Currently it tests both outputs (with and without `collapse_no_version`). I wanted to see the effect of this PR on both, so I added a test that generated both strings. The entire point is to be able to look at this diff https://github.com/pubgrub-rs/pubgrub/pull/163/commits/8b4378e6498567651440d8d7fc2dd262c181e8c8#diff-7781b2fddc85cc35c9f7b82403cc86e23906eb6eb1731a62af09e0659ef452bf . Now that I have documented that both outputs improved in ways I find acceptable, the test does not need to do both. Would you like me to remove the `collapse_no_version` one?

Reply:
Ah right, I don't know why I believed at some point the test was to compare the old and new errors, but there is no more "old" errors.

> The entire point is to be able to look at this diff https://github.com/pubgrub-rs/pubgrub/commit/8b4378e6498567651440d8d7fc2dd262c181e8c8#diff-7781b2fddc85cc35c9f7b82403cc86e23906eb6eb1731a62af09e0659ef452bf

Perfect! Makes sense to keep the test I guess. It can act as a snapshot test for future improvements or degradations of that error message.